src/physics/em/detail/MollerBhabhaInteractor.hh

```diff
@@ -7,67 +7,72 @@
 #include "physics/base/ParticleTrackView.hh"
 #include "physics/base/SecondaryAllocatorView.hh"
 #include "physics/base/Secondary.hh"
 #include "physics/base/Units.hh"
-#include "MollerBhabhaInteractorPointers.hh"
+#include "physics/material/ElementView.hh"
+
+#include "MollerBhabha.hh"
 
 namespace celeritas
 {
+namespace detail
+{
 //---------------------------------------------------------------------------//
 /*!
- * Brief class description.
+ * Perform Moller (e-e-) and Bhabha (e+e-) scattering.
  *
- * This is a model for XXXX process. Additional description
+ * This is a model for both Moller and Bhabha scattering processes.
  *
  * \note This performs the same sampling routine as in Geant4's
- * XXXX class, as documented in section XXX of the Geant4 Physics
- * Reference (release 10.6).
+ * G4MollerBhabhaModel class, as documented in section 10.1.4 of the Geant4
+ * Physics Reference (release 10.6).
  */
 class MollerBhabhaInteractor
 {
   public:
     // Construct with shared and state data
     inline CELER_FUNCTION
-    MollerBhabhaInteractor(const MollerBhabhaInteractorPointers& shared,
-                           const ParticleTrackView&              particle,
-                           const Real3&                          inc_direction,
-                           SecondaryAllocatorView&               allocate);
+    MollerBhabhaInteractor(const MollerBhabhaPointers& shared,
+                           const ParticleTrackView&    particle,
+                           const Real3&                inc_direction,
+                           SecondaryAllocatorView&     allocate);
 
     // Sample an interaction with the given RNG
     template<class Engine>
     inline CELER_FUNCTION Interaction operator()(Engine& rng);
 
-    //// COMMON PROPERTIES ////
-
-    //! Minimum incident energy for this model to be valid
-    static CELER_CONSTEXPR_FUNCTION units::MevEnergy min_incident_energy()
-    {
-        return units::MevEnergy{0}; // XXX
-    }
-
-    //! Maximum incident energy for this model to be valid
-    static CELER_CONSTEXPR_FUNCTION units::MevEnergy max_incident_energy()
-    {
-        return units::MevEnergy{0}; // XXX
-    }
-
   private:
     // Shared constant physics properties
-    const MollerBhabhaInteractorPointers& shared_;
-    // Incident gamma energy
+    const MollerBhabhaPointers& shared_;
+    // Incident energy
     const units::MevEnergy inc_energy_;
+    // Incident momentum
+    const units::MevMomentum inc_momentum_;
     // Incident direction
     const Real3& inc_direction_;
     // Allocate space for one or more secondary particles
     SecondaryAllocatorView& allocate_;
-};
+    // Incident particle flag for selecting Moller or Bhabha scattering
+    bool inc_particle_is_electron_;
+
+  private:
+    template<class Engine>
+    inline CELER_FUNCTION real_type sample_moller(Engine& rng);
+
+    template<class Engine>
+    inline CELER_FUNCTION real_type sample_bhabha(Engine& rng);
```

Comment:
These two guys don't call the allocator or otherwise change the internal class state, so they can be `const`.